tbx/core/utils/models.py

```diff
@@ -61,3 +62,42 @@ class SocialMediaSettings(BaseSiteSetting):
         default="{{ cookiecutter.project_name }}",
         help_text="Site name, used by Open Graph.",
     )
+
+
+class ColourTheme(models.TextChoices):
+    NONE = "", "None"
+    CORAL = "theme-coral", "Coral"
+    LAGOON = "theme-lagoon", "Lagoon"
+    BANANA = "theme-banana", "Banana"
+
+
+class ColourThemeMixin(models.Model):
+    """
+    Provides a `theme` field to allow pages to be styled with a colour theme.
+    """
+
+    theme = models.CharField(
+        max_length=25,
+        blank=True,
+        choices=ColourTheme.choices,
+        help_text=_(
+            "The theme will be applied to this page and all of it's "
+            "descendants. If no theme is selected, it will be derived from "
+            "this page's ancestors."
+        ),
+    )
+
+    @property
+    def theme_class(self):
+        if theme := self.theme:
+            return theme
+        if parent := self.get_parent():
+            specific_parent = parent.get_specific(deferred=True)
+            if not parent.is_root() and hasattr(specific_parent, "theme"):
+                return specific_parent.theme_class
```

Comment:
**Comment**: This could result in O(depth) queries. Would it be possible with something like `self.get_ancestors().specific().exclude(depth=1).exclude(theme=ColourTheme.NONE).last().theme`?

Reply:
I imagine not due to it not being a field on all Page classes.

Reply:
Thanks, I modified it slightly as follows

https://github.com/torchbox/torchbox.com/blob/b81fca07196cae4295adee401033b0808c91e32c/tbx/core/utils/models.py#L95-L102